modules/programs/mercurial.nix

```diff
@@ -0,0 +1,104 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.mercurial;
+
+in
+
+{
+
+  options = {
+    programs.mercurial = {
+      enable = mkEnableOption "Mercurial";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.mercurial;
+        defaultText = "pkgs.mercurial";
+        description = "Mercurial package to install.";
+      };
+
+      userName = mkOption {
+        type = types.str;
+        description = "Default user name to use.";
+      };
+
+      userEmail = mkOption {
+        type = types.str;
+        description = "Default user email to use.";
+      };
+
+      aliases = mkOption {
+        type = types.attrs;
+        default = {};
+        description = "Mercurial aliases to define.";
+      };
+
+      extraConfig = mkOption {
+        type = types.either types.attrs types.lines;
+        default = {};
+        description = "Additional configuration to add.";
+      };
+
+      iniContent = mkOption {
+        type = types.attrsOf types.attrs;
+        internal = true;
+      };
+
+      ignores = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        example = [ "*~" "*.swp" ];
+        description = "List of globs for files to be globally ignored.";
+      };
+
+      regexpIgnores = mkOption {
```

Comment:
I'd suggest to call this `ignoresRegexp` since the man page shows options in alphabetic order and I think it would be nice to have this option next to the `ignores` option.